### PR TITLE
Update tectonic version, enable M1 macs

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
- All dependencies are good to go for Apple silicon so fingers crossed (homebrew supports tectonic for M1 macs, so shouldn't be a problem)
- Update tectonic version to latest release

Close #2148, close #3704.

### CODE REVIEW

**Comment by staticfloat** [comment]
You shouldn't need to pass `--target` to `cargo`; it should work automatically, just like `gcc`.

**Comment by jeremiahpslewis** [comment]
@staticfloat In theory, yep, but in practice, I cannot get it to work / build. Think it may have to do with an issue in the rust build script, see here: https://github.com/JuliaPackaging/Yggdrasil/issues/3618

**Comment by staticfloat** [comment]
Yeah, you're right.  Mosè was not joking when he told me that the rust shards were a total crapfest recently.

Let's try manually inserting `--target ${rust_target}`.

**Comment by jeremiahpslewis** [comment]
One thing that I wonder is whether we're going to end up debugging this recipe conditional on the somewhat older rust version in Yggdrasil and then update rust to then need to redo this recipe for the new version...have a gut sense that we might want to tame rust before touching this again, but I don't have any skills to donate in that direction, so feel free to disregard my hunch.

**Comment by staticfloat** [comment]
Yes, I agree, unfortunately.

File: T/tectonic/build_tarballs.jl [hunk 1]
```diff
@@ -49,11 +50,11 @@ dependencies = [
     Dependency("FreeType2_jll"),
     Dependency("Graphite2_jll"),
     Dependency("HarfBuzz_jll"),
-    Dependency(PackageSpec(; name="ICU_jll", version=v"67.1.0")),
+    Dependency("ICU_jll", v"69.1"),
```
Comment (giordano) [inline T/tectonic/build_tarballs.jl:53]:
```suggestion
    Dependency("ICU_jll"; compat="69.1"),
```

File: T/tectonic/build_tarballs.jl [hunk 2]
```diff
@@ -49,11 +50,11 @@ dependencies = [
     Dependency("FreeType2_jll"),
     Dependency("Graphite2_jll"),
     Dependency("HarfBuzz_jll"),
```
Comment (giordano) [inline T/tectonic/build_tarballs.jl:52]:
```suggestion
    Dependency("HarfBuzz_jll"; compat="2.8.1"),
```

**Comment by giordano** [comment]
I don't think we want to build a local copy of harbuzz since we have it already.

**Comment by jeremiahpslewis** [comment]
Looks like this is getting closer, thanks to @giordano’s heroic rust rebuild, x86 windows still failing

**Pull request merged**